enginetest/queries/script_queries.go

```diff
@@ -1993,6 +1993,221 @@ CREATE TABLE tab3 (
 			},
 		},
 	},
+	{
+		Name: "last_insert_id(default) behavior",
+		SetUpScript: []string{
+			"create table t (pk int primary key auto_increment, i int default 0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "insert into t(pk) values (default);",
+				Expected: []sql.Row{{types.OkResult{RowsAffected: 1, InsertID: 1}}},
+			},
+			{
+				Query: "select last_insert_id()",
+				Expected: []sql.Row{
+					{uint64(1)},
+				},
+			},
+			{
+				Query: "select * from t",
+				Expected: []sql.Row{
+					{1, 0},
+				},
+			},
+
+			{
+				Query:    "insert into t(pk) values (default), (default), (default), (default), (default);",
+				Expected: []sql.Row{{types.OkResult{RowsAffected: 5, InsertID: 2}}},
+			},
+			{
+				Query: "select last_insert_id()",
+				Expected: []sql.Row{
+					{uint64(2)},
+				},
+			},
+			{
+				Query: "select * from t",
+				Expected: []sql.Row{
+					{1, 0},
+					{2, 0},
+					{3, 0},
+					{4, 0},
+					{5, 0},
+					{6, 0},
+				},
+			},
+
+			{
+				Query:    "insert into t(pk) values (10), (default);",
+				Expected: []sql.Row{{types.OkResult{RowsAffected: 2, InsertID: 10}}},
```

Comment:
(minor) should `InsertID` be 11? I couldn't see where MySQL was returning this field, but it seems like it would be the same as `last_insert_id`? (all I saw MySQL return was the `RowsAffected`) 

Not a big deal, and I don't think we need to change it here, since it isn't really user facing, but figured I'd mention it.

Reply:
It definitely seems like it should be 11.
Fixing it breaks a bunch of tests, and I'm not sure if they are wrong or not.
Will look further into this in a future PR.